src/common/actions/repositories.js

```diff
@@ -53,8 +54,124 @@ export function fetchUserRepositories(pageNumber) {
 
 export function fetchRepositoriesError(error) {
   return {
-    type: FETCH_REPOSITORIES_ERROR,
+    type: REPOSITORIES_FAILURE,
     payload: error,
     error: true
   };
 }
+
+// XXX may need to split out the fetch from the success and failure dispatch
+// to ensure than setRepositories happens in sync with refreshed user snaps
+export function fetchUserRepositoriesAndSnaps(owner) {
+  return (dispatch) => {
+    return Promise.all([
+      dispatch(fetchUserRepositories()),
+      dispatch(fetchUserSnaps(owner))
+    ]);
+  };
+}
+
+// XXX move to repository actions?
+// could be plural or singular, could be in one module or two, might be clearer in one for now
+export const REPOSITORY_TOGGLE_SELECT = 'REPOSITORY_TOGGLE_SELECT';
+
+export function toggleRepositorySelection(id) {
+  return {
+    type: REPOSITORY_TOGGLE_SELECT,
+    payload: {
+      id
+    }
+  };
+}
+
+export const REPOSITORY_BUILD = 'REPOSITORY_BUILD';
+export const REPOSITORY_SUCCESS = 'REPOSITORY_SUCCESS';
+export const REPOSITORY_FAILURE = 'REPOSITORY_FAILURE';
+export const REPOSITORY_RESET = 'REPOSITORY_RESET';
+
+export function buildRepositories(repositories) {
+  return (dispatch) => {
+    const promises = repositories.map(
+      (repository) => {
+        return dispatch(buildRepository(repository));
+      }
+    );
+    return Promise.all(promises).then(() => dispatch(push('/dashboard')));
+  };
+}
+
+export function buildRepository(repository) {
+  const { id, url, owner, name } = repository;
+
+  return async (dispatch) => {
```

Comment:
Isn't the promise-based orchestrateFoo do the same?

```
function orchestrateFoo() {
  return dispatch(fetchFoo()
    .then(
      thing => dispatch(fooSuccess(thing)),
      error => dispatch(fooFailure(error())
  )
)}
```

You specify rejected handler here, which just dispatches an action, so no other error or rejected promise is returned. What is redurned by `.then` is always a resolved promise, because any errors that may cause `fetchFoo` to reject will be caught by error handler in the same way try-catch would catch in in async-await).